### PR TITLE
[pytorch] Fix signed random_

### PR DESCRIPTION
**Pull request by albanD**
fix for #6338 and tests

### CODE REVIEW

**Review by ssnl — COMMENTED** [review]
Thanks Alban! Can you move the cuda part of the test to `test_cuda.py`? They can call the same staticmethod on TestTorch. :)

File: test/test_cuda.py [hunk 1]
```diff
@@ -1563,6 +1563,9 @@ def test_nvtx(self):
         torch.cuda.nvtx.mark("bar")
         torch.cuda.nvtx.range_pop()
 
+    def test_random_neg_values(self):
```
Comment (SsnL) [inline test/test_cuda.py]: skipif `not TEST_CUDA` please!

Comment (albanD) [reply test/test_cuda.py]: test_cuda is not assuming cuda is available? My bad :/

Comment (albanD) [reply test/test_cuda.py]: The tests that use the same logic just above are not protected like [this](https://github.com/pytorch/pytorch/blob/aa99aa1cb81fa4f5ff98785fcaba2cb88bedb57f/test/test_cuda.py#L1434-L1435) one.

**Review by ssnl — APPROVED** [review]
Good to merge once CI passes

**Review by soumith — APPROVED** [review]

**Pull request merged**